### PR TITLE
Add YCbCr Sampler Array Support

### PR DESCRIPTION
**Pull request by jiaxie-amd**
- Support ycbcr sampler array with constant indexes

ycbcr.conversion.*.sampler_array.*  CTS result:
-   Passed:        304/736 (41.3%)
-   Failed:        0/736 (0.0%)
-   Not supported: 432/736 (58.7%)
-   Warnings:      0/736 (0.0%)

Signed-off-by: Jiajun Xie <jiaxie@amd.com>

### CODE REVIEW

**Comment by jiaxie-amd** [comment]
This PR should be rebased on https://github.com/GPUOpen-Drivers/llpc/pull/598

**Comment by amdvlk-admin** [comment]
### Test summary for commit 1dd6cdc8288e08de6333ae19227e0570ad0c1ef3
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev e2f8eb0ffe21751b300995e795965d37e8d6dd67</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev 2c44bbc7b58efba12c501f71f5b268f83c1bdad5</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev f02738a5cf6c158c8c5b25a5048951f3059cabe2</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 958/958 (100.0%)</li>
<li>Failed: 0/958 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103143/103143 (100.0%)</li>
<li>Failed: 0/103143 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/PatchDescriptorLoad.cpp [hunk 1]
```diff
@@ -243,7 +243,11 @@ Value *PatchDescriptorLoad::getDescPtrAndStride(ResourceNodeType resType, unsign
 
     // We need to change the stride to 4 dwords. It would otherwise be incorrectly set to 12 dwords
     // for a sampler in a combined texture.
-    stride = builder.getInt32(DescriptorSizeSampler);
+    if (node->type == ResourceNodeType::DescriptorYCbCrSampler)
```
Comment (amdrexu) [inline lgc/patch/PatchDescriptorLoad.cpp:246]: Could be a more concise way:)
stride = builder.getInt32(node->type == ResourceNodeType::DescriptorYCbCrSampler ? DescriptorSizeSamplerYCbCr :  DescriptorSizeSampler);

File: lgc/builder/ImageBuilder.cpp [hunk 2]
```diff
@@ -750,16 +759,19 @@ Value *ImageBuilder::CreateImageSampleConvertYCbCr(Type *resultTy, unsigned dim,
                                                    Value *convertingSamplerDesc, ArrayRef<Value *> address,
                                                    const Twine &instName) {
   Value *pResult = nullptr;
```
Comment (amdrexu) [inline lgc/builder/ImageBuilder.cpp:761]: pResult -> result.

**Comment by jiaxie-amd** [comment]
> In this case, separating it out again becomes a bit complicated due to the change in convertingSamplerDesc, plus the Github infrastructure is not yet ideal for working this way, so it shouldn't hold up this commit here. I'm just explaining it as a matter of general principle.

You're right, I'll keep the principle :)

**Comment by amdvlk-admin** [comment]
### Test summary for commit ede74b90c0ea28c1a5cd1843d8349455948c9ed3
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev e2f8eb0ffe21751b300995e795965d37e8d6dd67</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev 2c44bbc7b58efba12c501f71f5b268f83c1bdad5</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev f02738a5cf6c158c8c5b25a5048951f3059cabe2</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 958/958 (100.0%)</li>
<li>Failed: 0/958 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103143/103143 (100.0%)</li>
<li>Failed: 0/103143 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 35490/35491 (100.0%)</li>
<li><details><summary>Failed: 1/35491 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.binding_model.descriptorset_random.sets32.constant.ubolimitlow.sbolimithigh.imglimithigh.noiub.uab.frag.ialimithigh.0
Stack trace: Timeout or crahsed
```
</p></details></li>


</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by jiaxie-amd** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit ede74b90c0ea28c1a5cd1843d8349455948c9ed3
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev e2f8eb0ffe21751b300995e795965d37e8d6dd67</li>
<li>SPVGEN: dev e843618d71c8263030551e7692b7683f20f627fe</li>
<li>XGL: dev 2c44bbc7b58efba12c501f71f5b268f83c1bdad5</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev f02738a5cf6c158c8c5b25a5048951f3059cabe2</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 958/958 (100.0%)</li>
<li>Failed: 0/958 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103143/103143 (100.0%)</li>
<li>Failed: 0/103143 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**